integration_tests/e2e/personal/edit/editWeight.cy.ts

```diff
@@ -0,0 +1,127 @@
+import { Role } from '../../../../server/data/enums/role'
+import EditWeight from '../../../pages/editPages/weight'
+import { editPageTests } from './editPageTests'
+
+context('Edit height (metric)', () => {
+  const prisonerNumber = 'G6123VU'
+  const bookingId = 1102484
+
+  editPageTests<EditWeight>({
+    prisonerNumber,
+    bookingId,
+    testSetup: () => {
+      cy.task('reset')
+      cy.setupUserAuth({
+        caseLoads: [
+          {
+            caseLoadId: 'MDI',
+            currentlyActive: true,
+            description: '',
+            type: '',
+            caseloadFunction: '',
+          },
+        ],
+        roles: [Role.PrisonUser, 'DPS_APPLICATION_DEVELOPER'],
+      })
+      cy.setupBannerStubs({ prisonerNumber, bookingId })
+      cy.task('stubPrisonPerson', { prisonerNumber })
+      cy.task('stubPrisonPersonUpdatePhysicalAttributes', { prisonerNumber })
+      cy.setupPersonalPageSubs({ prisonerNumber, bookingId })
+      cy.task('stubPersonalCareNeeds')
+    },
+    editUrl: `prisoner/${prisonerNumber}/personal/edit/weight`,
+    validInputs: { textInputs: { editField: '65' } },
+    invalidResponses: [
+      {
+        testDescription: 'Empty',
+        inputs: { textInputs: { editField: '' } },
+        errorMessages: ['Enter a number greater than 0'],
+      },
+      {
+        testDescription: 'Negative number',
+        inputs: { textInputs: { editField: '-10' } },
+        errorMessages: ['Enter a number greater than 0'],
+      },
+      {
+        testDescription: 'Zero',
+        inputs: { textInputs: { editField: '0' } },
+        errorMessages: ['Enter a number greater than 0'],
+      },
+    ],
+    editPageWithTitle: EditWeight,
+    editPageTitle: 'Edit weight',
+    successfulFlashMessage: 'Weight edited',
+  })
+})
+
+context('Edit weight (Imperial)', () => {
+  const prisonerNumber = 'G6123VU'
+  const bookingId = 1102484
+
+  editPageTests<EditWeight>({
+    prisonerNumber,
+    bookingId,
+    testSetup: () => {
+      cy.task('reset')
+      cy.setupUserAuth({
+        caseLoads: [
+          {
+            caseLoadId: 'MDI',
+            currentlyActive: true,
+            description: '',
+            type: '',
+            caseloadFunction: '',
+          },
+        ],
+        roles: [Role.PrisonUser, 'DPS_APPLICATION_DEVELOPER'],
+      })
+      cy.setupBannerStubs({ prisonerNumber, bookingId })
+      cy.task('stubPrisonPerson', { prisonerNumber })
+      cy.task('stubPrisonPersonUpdatePhysicalAttributes', { prisonerNumber })
+      cy.setupPersonalPageSubs({ prisonerNumber, bookingId })
+      cy.task('stubPersonalCareNeeds')
+    },
+    editUrl: `prisoner/${prisonerNumber}/personal/edit/weight/imperial`,
+    validInputs: { textInputs: { stones: '5', pounds: '3' } },
+    invalidResponses: [
+      // {
```

Comment:
I'm updating the validation in a subsequent PR so i've commented these out for now